### PR TITLE
Improved functions

### PR DESCRIPTION
**Pull request by GerardSmit**
Functions are now declered in seperated classes instead of the library itself. Whenever the user calls a function with too many (or few) arguments the warning will now display the position of the error.

### CODE REVIEW

File: src/MiScript.Tests/Basic/004_Warning.mst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿--SCRIPT--
+stop "world"
+"Test"
```
Comment (velddev) [inline src/MiScript.Tests/Basic/004_Warning.mst:3]: Why does this run test?

File: src/MiScript.Tests/Functions/001_Upper.mst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿--SCRIPT--
+var var = upper "foo"
+say $var
+say upper("bar")
```
Comment (velddev) [inline src/MiScript.Tests/Functions/001_Upper.mst:4]: love the idea of functions

File: src/MiScript/Ast/Expressions/CallExpression.cs [hunk 3]
```diff
@@ -1,12 +1,13 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
+using Miki.Localization;
 using MiScript.Models;
 
 namespace MiScript.Ast
 {
-    public class CallStatement : Statement
+    public class CallExpression : Expression
```
Comment (velddev) [inline src/MiScript/Ast/Expressions/CallExpression.cs:8]: What's the reason for the change?

**Review by velddev — APPROVED** [review]
Some comments but looking good

File: src/MiScript.Tests/Basic/004_Warning.mst (hunk 1 above)
Comment (GerardSmit) [reply src/MiScript.Tests/Basic/004_Warning.mst:3]: I was testing the warnings. In the test runner you get the warnings in the output:

image

In the future I want to check these warnings as well.

File: src/MiScript/Ast/Expressions/CallExpression.cs (hunk 3 above)
Comment (GerardSmit) [reply src/MiScript/Ast/Expressions/CallExpression.cs:8]: Because function calls can now return a value (e.g. `say upper("bar")`), because of this they are expressions now.

**Comment by GerardSmit** [comment]
Merged in the MikiStd repository.
